src/api/middlewares/isAuth.ts

```diff
@@ -1,33 +1,70 @@
-import jwt from 'express-jwt';
+// Imports
+import jwt from 'jsonwebtoken'
+
+// App Imports
 import config from '../../config';
+import User from '../../models/user';
+
+// Authentication middleware
+export default async function(request, response, next) {
+  let header = request.headers.authentication;
 
-/**
- * We are assuming that the JWT will come in a header with the form
- *
- * Authorization: Bearer ${JWT}
- *
- * But it could come in a query parameter with the name that you want like
- * GET https://my-bulletproof-api.com/stats?apiKey=${JWT}
- * Luckily this API follow _common sense_ ergo a _good design_ and don't allow that ugly stuff
- */
-const getTokenFromHeader = req => {
-  /**
-   * @TODO Edge and Internet Explorer do some weird things with the headers
-   * So I believe that this should handle more 'edge' cases ;)
-   */
-  if (
-    (req.headers.authorization && req.headers.authorization.split(' ')[0] === 'Token') ||
-    (req.headers.authorization && req.headers.authorization.split(' ')[0] === 'Bearer')
-  ) {
-    return req.headers.authorization.split(' ')[1];
+  if (header && header !== null) {
+    try {
+      const token = header.split(' ');
+      const userToken = jwt.verify(token[1], config.jwtSecret);
+      let user = await User.findOne({ _id: userToken.id });
+
+      if (user) {
+        request.auth = {
+          isAuthenticated: true,
+          user,
+        }
+      }
```

Comment:
Should there be an else to catch if the user does not/ no longer exists. We are getting a jwt that is signed but with an `id` that is no longer in the system. The other block where `isAuthenticated: false` will not be called since that is in the `else` statement for the check for headers.

Also the `catch` block for detecting an invalid token should be handled as well? Maybe return a direct error to the user?